torch/distributed/nn/api/remote_module.py

```diff
@@ -0,0 +1,129 @@
+#!/usr/bin/python3
+import uuid
+from typing import Any, Callable, Dict, Tuple
+
+import torch
+import torch.distributed.rpc as rpc
+from torch.distributed.nn.jit import instantiator
+
+
+def _gen_global_unique_name():
+    return f"{uuid.uuid4().hex}"
+
+
+def _instantiate_template(module_interface_cls, is_scriptable):
+    # Instantiate _RemoteModule class template on the local side.
+    generated_module = instantiator.instantiate_remote_module_template(
+        module_interface_cls, is_scriptable
+    )
+
+    remote_module_cls = generated_module._RemoteModule
+    remote_module_interface_cls = generated_module._RemoteModuleInterface
+
+    return remote_module_cls, remote_module_interface_cls
+
+
+def _script_module_creator_wrapper(module_creator, module_interface_cls, args, kwargs):
+    module = module_creator(*args, **kwargs)
+    script_module = torch.jit.script(module)
+    return rpc.RRef(script_module, module_interface_cls)
+
+
+def RemoteModule(
+    to: str,
+    module_creator: Callable,
+    args: Tuple = None,
+    kwargs: Dict[str, Any] = None,
+    global_unique_name: str = None,
+    module_interface_cls: Any = None,
+):
+    """
+        A RemoteModule instance can only be created after RPC initialization.
+        It creates a user-specified module on a specified remote node.
+        It behaves like a regular nn.Module except that the forward method is
+        executed on the remote node.
+        It takes care of autograd recording to ensure the backward pass propogates
+        gradients back to the corresponding remote module.
+        The arguments of foward_async and foward they take are the same
+        as the forward method of the module returned by the module_creator.
```

Comment:
double quote for `module_creator`, `foward_async` and `forward`

Reply:
Adding.